# Question 2: If Quadeca’s music did change, did the success of *Insecure* influence Quadeca to make music that reflected mainstream musical themes and styles?

## Setup

In [7]:
import os
import re
import csv
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from nltk import tokenize
from nltk.sentiment.vader import SentimentIntensityAnalyzer

In [8]:
# import notebooks to run
%run /Commjhub/jupyterhub/comm318_fall2019/raymon-shi/comm313_S21/comm313_S21_Final_Project/data/functions.ipynb

In [9]:
%store -r after_all_lyrics
%store -r all_rap_chart_lyrics
sid = SentimentIntensityAnalyzer()

## Sentiment Analysis Across Both Corpuses

### Quadeca's Music

In [41]:
print('the vader polarity average scores for each song in the corpus are:')
quad_vps_list = print_polarity_score_album(after_all_lyrics)
quad_vps_list

the vader polarity average scores for each song in the corpus are:


[0.9881,
 0.9975,
 0.9827,
 0.9729,
 0.9967,
 0.9905,
 0.9973,
 -0.658,
 -0.9987,
 0.9926,
 -0.3506,
 -0.8705,
 0.9902,
 0.9091,
 0.6872,
 0.9664,
 -0.9801,
 0.9979,
 -0.5204,
 0.9768,
 -0.9829,
 -0.9878,
 -0.8512,
 0.4019,
 -0.942,
 0.9166,
 0.4153,
 -0.9688,
 -0.9463,
 0.9883,
 0.1027]

In [42]:
quad_pos_songs = list()
quad_neg_songs = list()
quad_neu_songs = list()

sort_sentiment_songs(quad_vps_list, quad_pos_songs, quad_neg_songs, quad_neu_songs)

There are 19 positive songs in this album
There are 12 negative songs in this album
There are 0 neutral songs in this album


In [34]:
quad_vps = get_polarity_score(after_all_lyrics)

print('the vader polarity sccore is:', quad_vps)

the vader polarity sccore is: 0.20043225806451603


#### Overall Observation from Quadeca's Music

According to the vader polarity scores, there are a total of 19 positive songs, 12 negative songs, and 0 neutral songs. This means that there are 61.29% positive songs, 38.71% negative songs, and 0.00% neutral songs. The overall corpus has a vader polarity score of 0.200 and since this is greater than or equal to 0.05, that means that the overall corpus is a positive corpus. However, since this number is closer to 0.00, that means that the corpus is only slightly positive.

Another thing to note is that this polarity score dropped from the previous calculate polarity score to 0.237 to the new polarity score of 0.200. This is probably due to the fact that the new polarity score is calculated across the entirety of Quadeca's later albums rather than doing each album individually. Although there is a drop in polarity score, the polarity score is still in the same range. 

### Billboard Rap Music

In [35]:
print('the vader polarity average scores for each song in the corpus are:')
rap_vps_list = print_polarity_score_album(all_rap_chart_lyrics)
rap_vps_list

the vader polarity average scores for each song in the corpus are:


[0.8948,
 0.9988,
 -0.9723,
 -0.9771,
 -0.8081,
 0.9735,
 0.9431,
 0.9918,
 0.9033,
 0.3098,
 -0.9953,
 0.992,
 0.9985,
 -0.8659,
 0.9978,
 -0.9994,
 -0.9972,
 0.9954,
 -0.9752,
 0.9997,
 0.9976,
 0.9947,
 -0.9711,
 -0.9995,
 0.4954,
 0.999,
 -0.9883,
 0.9694,
 0.9994,
 -0.9995,
 -0.9916,
 -0.9941,
 -0.9731,
 -0.9981,
 -0.9877,
 0.5345,
 0.2901,
 -0.997,
 -0.9986,
 -0.9948,
 -0.9501,
 -0.7452,
 -0.9516,
 -0.9949,
 -0.9968,
 0.9993,
 -0.9869,
 -0.99,
 0.9957,
 0.2654,
 0.7936,
 -0.9738,
 0.8704,
 0.9995,
 0.9954,
 -0.9437,
 -0.9403,
 -0.9786,
 0.9877,
 0.9585,
 -0.8404,
 -0.9517,
 -0.9986,
 0.9709,
 -0.9977,
 -0.9712,
 0.8332,
 -0.9958,
 -0.9969,
 0.9911,
 -0.99,
 -0.9965,
 0.9982,
 0.4027,
 -0.9967,
 -0.9749,
 -0.9997,
 -0.9998,
 0.9918,
 0.985,
 0.9961,
 -0.9942,
 -0.9518,
 -0.9983,
 -0.9997,
 0.9431,
 -0.9996,
 -0.9573,
 -0.9894,
 0.4462,
 -0.9992,
 -0.6278,
 0.9747,
 -0.979,
 0.3494,
 -0.999,
 -0.9971,
 -0.9604,
 -0.965,
 -0.9977,
 -0.9905,
 0.9783,
 -0.9866,
 0.9964,
 -0.7254,
 -0.

In [37]:
rap_pos_songs = list()
rap_neg_songs = list()
rap_neu_songs = list()

sort_sentiment_songs(rap_vps_list, rap_pos_songs, rap_neg_songs, rap_neu_songs)

There are 60 positive songs in this album
There are 89 negative songs in this album
There are 1 neutral songs in this album


In [39]:
rap_vps = get_polarity_score(all_rap_chart_lyrics)

print('the vader polarity sccore is:', rap_vps)

the vader polarity sccore is: -0.22581466666666666


#### Overall Observations from Billboard Rap Music

According to the vader polarity scores, there are a total of 60 positive songs, 89 negative songs, and 1 neutral song. This means that there are 40.00% positive songs, 59.33% negative songs, and 0.66% neutral songs. The overall corpus has a vader polarity score of -0.226 and since this is less than or equal to -0.05, this means that the overall corpus is a negative corpus. However, since the number is closer to 0.00, this means that the corpus is only slightly negative.

### Conclusion From Sentiment Analysis

After concluding a sentiment analysis across both corpuses, I found that the sentiment scores between Quadeca's music and the Billboard Rap music is fairly different. In fact, the sentiment scores of the two corpuses are nearly an absolute value away from each other. Quadeca's music was found to have a slightly positive sentiment with an average polarity score of 0.200, while the Billboard Rap music was found to have a slightly negative sentiment with an average polarity score of -0.226.

This further shows that Quadeca's music was not influenced by the mainstream rap music nor does it reflect the same sentiment as the mainstream rap music.

Although, I am surprised to see that the overall sentiment score of the mainstream rap music was negative. From the topics that I discovered through the KWIC analysis, I thought that the mainstream rap music would have a more positive trend and ironically, I originally thought that Quadeca's music was going to have a more negative trend.